src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunSymbolNodeFactory.cs

```diff
@@ -814,5 +821,52 @@ public override int GetHashCode()
             }
             return node;
         }
+
+        Dictionary<int, ObjectNode> _rvaFieldSymbols = new Dictionary<int, ObjectNode>();
+
+        public ObjectNode GetRvaFieldNode(FieldDesc fieldDesc)
+        {
+            Debug.Assert(fieldDesc.HasRva);
+            EcmaField ecmaField = (EcmaField)fieldDesc;
```

Comment:
This cast can fail if it's a field on a generic type. CoreCLR test collateral has coverage for this because [I've hit it in the past](https://github.com/dotnet/corert/pull/2228).